gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -145,10 +145,12 @@ public CloudStoragePath getPath(String first, String... more) {
    * Does nothing.
    */
   @Override
-  public void close() {}
+  public void close() throws IOException {
```

Comment:
It would actually be a significant refactor, because right now channels are created by the CloudStorageFileSystemProvider, completely bypassing the CloudStorageFileSystem. We'd probably want to think carefully before that refactor. And unless it's a requirement for merging into master, I'd rather first merge into master and then do it since every passing day means the codebases deviate a bit more.